controllers/wavefrontalert_controller.go

```diff
@@ -87,12 +92,35 @@ func (r *WavefrontAlertReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	if err := r.Get(ctx, req.NamespacedName, &wfAlert); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
+	var status alertmanagerv1alpha1.WavefrontAlertStatus
 	//Main responsibilities of the Wavefront Alert Controller
 	// Check if it is delete request
 	if !wfAlert.ObjectMeta.DeletionTimestamp.IsZero() {
+		requeueFlag := false
 		// Delete use case
-		return ctrl.Result{}, r.HandleDelete(ctx, &wfAlert)
+		if err := r.HandleDelete(ctx, &wfAlert); err != nil {
+			log.Error(err, "unable to delete the alert")
+			requeueFlag = true
+		}
+		return ctrl.Result{Requeue: requeueFlag}, nil
```

Comment:
its defined by underlying Kubebuilder implementation. It uses exponential back off too. We provide custom value in case of lets say you want this controller to verify the state from etcd and wavefront every 24 hours so that it will make sure etcd and wavefront are in sync